parquet-column/src/main/java/org/apache/parquet/column/statistics/BinaryStatistics.java

```diff
@@ -18,122 +18,259 @@
     } else {
-      updateStats(binaryStats.getMin(), binaryStats.getMax());
+      updateStatsSigned(binaryStats.genericGetMinSigned(), binaryStats.genericGetMaxSigned());
+      updateStatsUnsigned(binaryStats.genericGetMinUnsigned(), binaryStats.genericGetMaxUnsigned());
     }
   }
 
   /**
-   * Sets min and max values, re-uses the byte[] passed in.
-   * Any changes made to byte[] will be reflected in min and max values as well.
-   * @param minBytes byte array to set the min value to
-   * @param maxBytes byte array to set the max value to
+   * Sets minSigned and maxSigned values, re-uses the byte[] passed in.
+   * Any changes made to byte[] will be reflected in minSigned and maxSigned values as well.
+   * @param minBytes byte array to set the minSigned value to
+   * @param maxBytes byte array to set the maxSigned value to
    */
   @Override
   public void setMinMaxFromBytes(byte[] minBytes, byte[] maxBytes) {
-    max = Binary.fromReusedByteArray(maxBytes);
-    min = Binary.fromReusedByteArray(minBytes);
+    maxSigned = Binary.fromReusedByteArray(maxBytes);
+    minSigned = Binary.fromReusedByteArray(minBytes);
+    maxUnsigned = maxSigned.copy();
+    minUnsigned = minSigned.copy();
     this.markAsNotEmpty();
   }
 
   @Override
+  public void setMinMaxSignedFromBytes(byte[] minBytes, byte[] maxBytes) {
+    this.minSigned = Binary.fromReusedByteArray(minBytes);
+    this.maxSigned = Binary.fromReusedByteArray(maxBytes);
+  }
+
+  @Override
+  public void setMinMaxUnsignedFromBytes(byte[] minBytes, byte[] maxBytes) {
+    this.minUnsigned = Binary.fromReusedByteArray(minBytes);
+    this.maxUnsigned = Binary.fromReusedByteArray(maxBytes);
+  }
+
+  /**
+   * Use either getMaxBytesSigned() or getMaxBytesUnsigned() directly instead.
+   */
+  @Deprecated
+  @Override
   public byte[] getMaxBytes() {
-    return max == null ? null : max.getBytes();
+    return getMaxBytesSigned();
   }
 
+  /**
+   * Use either getMinBytesSigned() or getMinBytesUnsigned() directly instead.
+   */
+  @Deprecated
   @Override
   public byte[] getMinBytes() {
-    return min == null ? null : min.getBytes();
+    return getMinBytesSigned();
+  }
+
+  public byte[] getMaxBytesSigned() {
+    return maxSigned == null ? null : maxSigned.getBytes();
+  }
+
+  @Override
+  public byte[] getMinBytesSigned() {
+    return minSigned == null ? null : minSigned.getBytes();
+  }
+
+  @Override
+  public byte[] getMaxBytesUnsigned() {
+    return maxUnsigned == null ? null : maxUnsigned.getBytes();
+  }
+
+  @Override
+  public byte[] getMinBytesUnsigned() {
+    return minUnsigned == null ? null : minUnsigned.getBytes();
   }
 
   @Override
   public boolean isSmallerThan(long size) {
-    return !hasNonNullValue() || ((min.length() + max.length()) < size);
+    return !hasNonNullValue() || (((minSigned.length() + maxSigned.length()) < size) && ((minUnsigned.length() + maxUnsigned.length()) < size));
```

Comment:
seems like minSigned.length() + maxSigned.length() should be the same as minUnsigned.length() + maxUnsigned.length() right? we could think about dropping one pair from this comparison